### ISSUE TITLE
hrtime test fails on MacOS

### ISSUE THREAD

**Issue by basvandijk** [libuv/libuv#2342]
* **Version**: `libuv-1.29.1`
* **Platform**: `Darwin bassbox 18.6.0 Darwin Kernel Version 18.6.0: Thu Apr 25 23:16:27 PDT 2019; root:xnu-4903.261.4~2/RELEASE_X86_64 x86_64`

The `hrtime` tests fails on MacOS:

```
nix-build -A libuv
...
not ok 109 - hrtime
# timeout
# Output from process `hrtime`: (no output)
```

**Comment by bnoordhuis** [comment]
The test is somewhat timing sensitive. It normally takes about 3.5 seconds to run but I suppose it might hit the test timeout when run on a heavily loaded system.

Does it pass with this patch applied?
```diff
diff --git a/test/test-list.h b/test/test-list.h
index 5c1a1c99..a48f6f38 100644
--- a/test/test-list.h
+++ b/test/test-list.h
@@ -779,7 +779,7 @@ TASK_LIST_START
 
   TEST_ENTRY  (tmpdir)
 
-  TEST_ENTRY  (hrtime)
+  TEST_ENTRY_CUSTOM (hrtime, 0, 0, 10000)
 
   TEST_ENTRY_CUSTOM (getaddrinfo_fail, 0, 0, 10000)
   TEST_ENTRY_CUSTOM (getaddrinfo_fail_sync, 0, 0, 10000)
```

**Comment by bnoordhuis** [comment]
The test is somewhat timing sensitive. It normally takes about 3.5 seconds to run but I suppose it might hit the test timeout when run on a heavily loaded system.

Does it pass with this patch applied?
```diff
diff --git a/test/test-list.h b/test/test-list.h
index 5c1a1c99..a48f6f38 100644
--- a/test/test-list.h
+++ b/test/test-list.h
@@ -779,7 +779,7 @@ TASK_LIST_START
 
   TEST_ENTRY  (tmpdir)
 
-  TEST_ENTRY  (hrtime)
+  TEST_ENTRY_CUSTOM (hrtime, 0, 0, 10000)
 
   TEST_ENTRY_CUSTOM (getaddrinfo_fail, 0, 0, 10000)
   TEST_ENTRY_CUSTOM (getaddrinfo_fail_sync, 0, 0, 10000)
```

**Comment by bnoordhuis** [comment]
The test is somewhat timing sensitive. It normally takes about 3.5 seconds to run but I suppose it might hit the test timeout when run on a heavily loaded system.

Does it pass with this patch applied?
```diff
diff --git a/test/test-list.h b/test/test-list.h
index 5c1a1c99..a48f6f38 100644
--- a/test/test-list.h
+++ b/test/test-list.h
@@ -779,7 +779,7 @@ TASK_LIST_START
 
   TEST_ENTRY  (tmpdir)
 
-  TEST_ENTRY  (hrtime)
+  TEST_ENTRY_CUSTOM (hrtime, 0, 0, 10000)
 
   TEST_ENTRY_CUSTOM (getaddrinfo_fail, 0, 0, 10000)
   TEST_ENTRY_CUSTOM (getaddrinfo_fail_sync, 0, 0, 10000)
```

**Comment by basvandijk** [comment]
I can confirm this patch fixes the test. 

I also noticed that hydra (the Nix-based CI system) was [able to build `libuv`](https://hydra.nixos.org/job/nixpkgs/trunk/libuv.x86_64-darwin) some hours ago. So it indeed looks like a time sensitive test.

**Comment by bnoordhuis** [comment]
Thanks for confirming. This should be addressed when #2345 lands, hopefully in time for the next release.

**Comment by vtjnash** [comment]
Appears to still be failing quite often for me: https://ci.nodejs.org/job/libuv-test-commit-osx-cmake/160/nodes=osx1015/console
